includes/Checker/AJAX_Runner.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\AJAX_Runner
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+use WordPress\Plugin_Check\Utilities\Plugin_Request_Utility;
+
+/**
+ * AJAX Runner class.
+ *
+ * @since n.e.x.t
+ */
+class AJAX_Runner extends Abstract_Check_Runner {
+
+	/**
+	 * An instances of the Checks class.
+	 *
+	 * @since n.e.x.t
+	 * @var Checks
+	 */
+	protected $checks;
+
+	/**
+	 * Checks if the current request is an AJAX request for the Plugin Checker.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool
+	 */
+	public function is_plugin_check() {
+		if ( 0 !== strpos( $_SERVER['REQUEST_URI'], '/admin-ajax.php' ) ) {
+			return false;
+		}
+
+		if ( ! isset( $_REQUEST['action'] ) || 'plugin_check_run_checks' !== $_REQUEST['action'] ) {
+			return false;
+		}
+
+		if ( ! check_ajax_referer( 'plugin_check_run_checks' ) ) {
+			return false;
+		}
```

Comment:
This doesn't seem relevant for this method. Whether the referer / nonce are valid doesn't define whether it's a plugin check request.

This validation is critical of course, but only at the point where we validate the request, not to determine whether it _is_ a plugin check request or not.